[Tan Tag Han] iP

taggyhan opened the pull request:
# Yapper

Yapper is a task management application that helps you stay organized and keep track of your tasks. It provides a simple and intuitive interface for managing tasks efficiently.

## Features

- List tasks
- Mark tasks as done
- Add new tasks
- Delete tasks
- Search for tasks

## Getting Started

To get started with Yapper, follow these steps:

1. Download the Yapper application from [here](https://github.com/taggyhan/ip).
2. Double-click the downloaded file to launch the application.
3. Start adding your tasks and let Yapper manage them for you. 😊

## Usage

Yapper supports various commands to interact with your tasks. Here are some examples:

- To list all tasks, use the command `list`.
- To mark a task as done, use the command `mark <taskIndex>`.
- To add a new task, use the command `todo <taskDescription>`.
- To delete a task, use the command `delete <taskIndex>`.
- To search for tasks, use the command `find <keyword>`.

## Example

Here's an example of how to use Yapper in your Java program:

- [x] Completed task
- [ ] Incomplete task

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

taggyhan committed:
Add .gitignore to ignore .class files

taggyhan committed:
Level-0: Rename main class to Yapper and add greeting and exit functionality

taggyhan committed:
Level-1: Implement echo feature and exit on 'bye' command

taggyhan committed:
Level-2: Add ability to store and list tasks

taggyhan committed:
Level-3: Add ability to mark tasks as done and not done

taggyhan committed:
Level-4: Add support for Todo, Deadline, and Event tasks

taggyhan committed:
Update EXPECTED.TXT for A-TextUiTesting to match program output

taggyhan committed:
Level-5: Add exception handling for incorrect inputs and missing descriptions

taggyhan committed:
Level-6: Implement task deletion with ArrayList

taggyhan committed:
Refactor Yapper.java

taggyhan committed:
redid level-5

taggyhan committed:
Use enums for command types

taggyhan committed:
Added Character to my bot

taggyhan committed:
Add tag for automated text UI testing

taggyhan committed:
Implement  Level 7

taggyhan committed:
Merge branch-Level-7 implementing task persistence (Level 7)

taggyhan committed:
Implement date and time handling

taggyhan committed:
Implement date and time handling for Level 8

taggyhan committed:
Merge branch-Level-8

taggyhan committed:
Refactor code

taggyhan committed:
Merge branch-A-MoreOOP implementing OOP refactoring

taggyhan committed:
Improve OOP

taggyhan committed:
Merge branch-A-MoreOOP with improved OOP structure

taggyhan committed:
Implement back custom exceptions

taggyhan committed:
Merge branch-A-MoreOOP with custom exceptions

taggyhan committed:
Refactor: Organize code into packages

taggyhan committed:
Merge branch-A-Packages into master

taggyhan committed:
Merge Gradle support into master

taggyhan committed:
add Gradle support

taggyhan committed:
,erge Gradle support

taggyhan committed:
Update .gitignore to exclude src/main/java/data directory

taggyhan committed:
Add JUnit tests for Todo class

taggyhan committed:
Add JUnit tests

taggyhan committed:
Merge branch-A-Junit with JUnit tests

taggyhan committed:
Change yapper.java to fix coding style

taggyhan committed:
Change parser.java to fix coding style

taggyhan committed:
Implement find function

taggyhan committed:
Fix Typo in Deadline.task

taggyhan committed:
Add JavaDocs

taggyhan committed:
Fix typo in Event and Deadline File

taggyhan committed:
Merge branch-A-JavaDoc into master

taggyhan committed:
Merge branch-A-CodingStandard into master

taggyhan committed:
Merge branch-Level-9 into master

taggyhan committed:
Standardize format

taggyhan committed:
Fix Error in Parser.java

oolongwater commented on src/main/java/yapper/main/Yapper.java:
```diff
@@ -0,0 +1,191 @@
+                CommandType command = Parser.parseCommand(fullCommand);
+
+                switch (command) {
+                case LIST:
+                    ui.printTasks(tasks);
+                    break;
+                case MARK:
+                    handleMark(fullCommand);
+                    break;
+                case UNMARK:
+                    handleUnmark(fullCommand);
+                    break;
+                case TODO:
+                    handleTodo(fullCommand);
+                    break;
+                case DEADLINE:
+                    handleDeadline(fullCommand);
+                    break;
+                case EVENT:
+                    handleEvent(fullCommand);
+                    break;
+                case DELETE:
+                    handleDelete(fullCommand);
+                    break;
+                case BYE:
+                    isExit = true;
+                    ui.printGoodbye();
+                    break;
```
Consider case XXX -> fn().
e.g: `case LIST -> handleList();`

taggyhan replied:
Sure , good suggestion

oolongwater commented on src/main/java/yapper/ui/Ui.java:
```diff
@@ -0,0 +1,98 @@
+        System.out.println("____________________________________________________________");
+    }
+
+    public void printLine() {
+        System.out.println("____________________________________________________________");
+    }
```
`printLine()` could be more descriptive. A name like `printSeparatorLine()` would better convey its purpose.

taggyhan replied:
thanks for the suggestion

oolongwater commented on src/main/java/yapper/ui/Ui.java:
```diff
@@ -0,0 +1,98 @@
+        System.out.println("____________________________________________________________");
+    }
+
+    public void printTaskRemoved(Task task, int taskCount) {
+        System.out.println("____________________________________________________________");
+        System.out.println(" Okay bOSS ur task got removed");
+        System.out.println("   " + task);
+        System.out.println(" Okay Boss! Now you have " + taskCount + " tasks in the list.");
+        System.out.println("____________________________________________________________");
+    }
```
There is inconsistency in the capitalization and formality of printed messages. The use of "bOSS" is inconsistent and unprofessional. It should be corrected to "Boss" for consistency and readability. The message "Okay bOSS ur task got removed" should be rephrased for professionalism and clarity. Consider using "Okay Boss, your task has been removed."

taggyhan replied:
Yes, this is very true

oolongwater commented on src/main/java/yapper/ui/Ui.java:
descriptive and used appropriate verbs, aligning with the naming conventions.

taggyhan replied:
thanks you

oolongwater reviewed (commented):
Code effectively handles user interactions by providing clear methods for printing various messages and handling user input. While the code follows good practices in terms of method naming and variable declarations, there are opportunities for improvement in naming conventions, such as using more descriptive names for the class itself and some methods.

Additionally, ensuring consistency in the tone and formatting of output messages, along with reducing code duplication, would enhance readability and maintainability. Overall, the class is well-structured, but minor refinements could improve its clarity and professionalism.

vinc3leong commented on src/main/java/yapper/storage/Storage.java:
```diff
@@ -0,0 +1,68 @@
+        this.filePath = filePath;
+    }
+
+    public ArrayList<Task> load() throws IOException {
```
The methods load() and save() could benefit from Javadoc comments.

taggyhan replied:
sure

vinc3leong commented on src/main/java/yapper/ui/Ui.java:
```diff
@@ -0,0 +1,98 @@
+package yapper.ui;
```
Everything else seems good. 
-adhered to the naming conventions
-overall tidy layout 
-consistent ordering of import statements

vinc3leong reviewed (commented):
Everything else seems good. :)
-adhered to the naming conventions
-overall tidy layout
-consistent ordering of import statements

taggyhan committed:
Fix style with checkstyle

Peanuts359 commented on src/main/java/yapper/main/Yapper.java:
```diff
@@ -0,0 +1,191 @@
+            }
+        }
+    }
+
```
These names make it much easier to tell what the methods do.

But I'd still add JavaDoc comments here if I were you.

Peanuts359 commented on src/main/java/yapper/task/TaskList.java:
```diff
@@ -0,0 +1,46 @@
+    public ArrayList<Task> getTasks() {
+        return tasks;
+    }
+
```
Straight to the point.

Peanuts359 commented on src/main/java/yapper/storage/Storage.java:
```diff
@@ -0,0 +1,68 @@
+        }
+        return tasks;
+    }
+
```
I think spacing out some sections will make the methods more readable.

Peanuts359 reviewed (commented):
Not bad.

taggyhan committed:
Add Gui

taggyhan committed:
Merging branch-A-CheckStyle into master

taggyhan committed:
Merging branch-Level-10 into master

taggyhan committed:
Fix style and exit function

taggyhan committed:
Fix typo in Yapper.java

taggyhan committed:
Refactor Code

taggyhan committed:
fix unknowncommandexception

taggyhan committed:
clean code and fix style

taggyhan committed:
fix style more

taggyhan committed:
Fix style 3.0

taggyhan committed:
Improve Storage.java

taggyhan committed:
Add assertions to task operations in Yapper

Assertions were added to various task operations within the Yapper
task management application to document assumptions and improve code
quality.

These assertions ensure:
- A task is marked as done after the "mark" operation.
- A task is marked as not done after the "unmark" operation.
- Tasks are correctly added to the task list during todo, event, and deadline operations.
- Inputs provided to the command methods are non-null.

Including these assertions helps prevent bugs

taggyhan committed:
Add more exceptions

taggyhan committed:
Refactor command handling logic with helper function

Introduced a new helper function splitAndValidateCommand to handle
command splitting and validation. This function is used to split the
user's full command string and validate the number of parts.

This change was made to remove repetitive logic from the command
handlers, such as handleMark, handleDelete, and handleUnmark.
Previously, each of these methods handled command parsing manually.

By centralizing the command parsing and validation into a helper
function, the code is now easier to maintain and more readable.
Furthermore, this change reduces the likelihood of inconsistent error
handling between methods.

- Refactored command parsing into splitAndValidateCommand
- Removed repetitive code from command handlers
- Improved code maintainability and consistency

taggyhan committed:
Merge pull request #1 from taggyhan/branch-A-Assertions

Add assertions to improve code quality

taggyhan committed:
Merge pull request #2 from taggyhan/branch-A-CodeQuality

Refactor command handling logic with helper function

taggyhan committed:
Merge branch 'master' into branch-A-CodeQuality
add code quality

taggyhan committed:
Add GitHub Actions CI for Java project

taggyhan committed:
Merge branch 'branch-A-CI'

taggyhan committed:
update yml

taggyhan committed:
update yml

taggyhan committed:
Implement snooze feature for deadline tasks

Add the ability to snooze or postpone deadline tasks by a specified
amount of time (either days or hours). This feature enhances the usability
of the application by making it more flexible for users to manage deadlines.

Changes include:
- A new snooze command for deadlines, allowing users to delay tasks by days or hours.
- Updated Deadline.java to incorporate the snooze functionality.
- Added exception handling to ensure correct snooze format ('d' for days, 'h' for hours).

Why this is important:
- This feature provides better task management as users can defer tasks without needing
  to manually remove and recreate deadlines.
- It adds flexibility, ensuring that deadlines can be shifted without hassle.

taggyhan committed:
Merge branch-BCD-Extension: Implement snooze functionality for deadlines

Integrate the snooze feature into master. Users can now postpone deadlines
by days or hours, allowing better flexibility and control over tasks with deadlines.

taggyhan committed:
Enhance Java basics using Copilot

taggyhan committed:
Make find command case-insensitive

taggyhan committed:
Update GUI to show Yapper

taggyhan committed:
Improve UI with yapper name

taggyhan committed:
Fix Style

taggyhan committed:
Fix Gitflow

taggyhan committed:
Fix Test Files

taggyhan committed:
Fix Test Files further

taggyhan committed:
Fix Style

taggyhan committed:
Fix Style

taggyhan committed:
Fix workflow and style

taggyhan committed:
Fix workflow and style

taggyhan committed:
Merge branch 'branch-cicdfix'

Fix CI and tests

taggyhan committed:
Add test

taggyhan committed:
Fix Tests

taggyhan committed:
Updated user guide for Yapper

taggyhan committed:
Merge branch-A-UserGuide into master

taggyhan committed:
Add Ui.png

taggyhan committed:
Fix typo in readme

taggyhan committed:
Merge branch-A-UserGuide into master

taggyhan committed:
Update Javadocs
